### PR TITLE
[GEOS-8110] Fix IndexOutOfBounds in WMSWorkspaceQualifier

### PR DESCRIPTION
**Pull request by tbarsballe**
https://osgeo-org.atlassian.net/browse/GEOS-8110

### CODE REVIEW

**Comment by dvntucker** [comment]
Looks fine, but can we use something like this instead?

http://docs.oracle.com/javase/8/docs/api/java/lang/String.html#join-java.lang.CharSequence-java.lang.Iterable-

**Comment by tbarsballe** [comment]
Yes, that's a much better way of doing this. Updated.

File: src/wms/src/test/java/org/geoserver/wms/wms_1_3/GetMapIntegrationTest.java [hunk 1]
```diff
@@ -373,7 +373,17 @@ public void testLayerGroupModeEo() throws Exception {
...
         }
-    }   
+    }
+
+    @Test
+    public void testWorkspaceQualifiedNoLayersError() throws Exception {
+        Document dom = getAsDOM("sf/wms?bbox=" + bbox + "&styles="
+                + "&layers=" + "&format=image/png" + "&request=GetMap" + "&width=550"
+                + "&height=250" + "&srs=EPSG:4326", Charset.defaultCharset().displayName());
+        print(dom);
```
Comment (aaime) [inline src/wms/src/test/java/org/geoserver/wms/wms_1_3/GetMapIntegrationTest.java]: Leftover print

**Comment by tbarsballe** [comment]
Backported to 2.11.x and 2.10.x

**Pull request merged**